File: Week -3/Qs8.c
#include <stdio.h>

// Structure to represent the date of joining
typedef struct {
    int day;
    int month;
    int year;
} Date;

// Structure to represent an employee
typedef struct {
    int employeeCode;
    char employeeName[50];
    float salary;
    int departmentNumber;
    Date dateOfJoining;
} Employee;

// Function to input employee information
void inputEmployee(Employee *e) {
    printf("Enter employee code: ");
    scanf("%d", &e->employeeCode);
    
    printf("Enter employee name: ");
    scanf(" %[^\n]%*c", e->employeeName); // Read string with spaces
    
    printf("Enter salary: ");
    scanf("%f", &e->salary);
    
    printf("Enter department number: ");
    scanf("%d", &e->departmentNumber);
    
    printf("Enter date of joining (day month year): ");
    scanf("%d %d %d", &e->dateOfJoining.day, &e->dateOfJoining.month, &e->dateOfJoining.year);
}

// Function to display employee information
void displayEmployee(const Employee *e) {
    printf("\nEmployee Code: %d\n", e->employeeCode);
    printf("Employee Name: %s\n", e->employeeName);
    printf("Salary: %.2f\n", e->salary);
    printf("Department Number: %d\n", e->departmentNumber);
    printf("Date of Joining: %02d/%02d/%04d\n", e->dateOfJoining.day, e->dateOfJoining.month, e->dateOfJoining.year);
}

// Main function
int main() {
    Employee employees[10];
    
    // Input information for 10 employees
    for (int i = 0; i < 10; i++) {
        printf("\nEnter information for employee %d:\n", i + 1);
        inputEmployee(&employees[i]);
    }
    
    // Display information for employees with salary >= 10000
    printf("\nEmployees with salary >= 10000:\n");
    for (int i = 0; i < 10; i++) {
        if (employees[i].salary >= 10000) {
            displayEmployee(&employees[i]);
        }
    }
    
    return 0;
}

File: Week -3/Qs8.py
class Date:
    def __init__(self, day=1, month=1, year=2020):
        self.day = day
        self.month = month
        self.year = year

    def __str__(self):
        return f"{self.day:02d}/{self.month:02d}/{self.year}"

class Employee:
    def __init__(self, employee_code=0, name="", salary=0.0, department_number=0, date_of_joining=None):
        self.employee_code = employee_code
        self.name = name
        self.salary = salary
        self.department_number = department_number
        self.date_of_joining = date_of_joining if date_of_joining is not None else Date()

    def __str__(self):
        return (f"Employee Code: {self.employee_code}\n"
                f"Employee Name: {self.name}\n"
                f"Salary: {self.salary:.2f}\n"
                f"Department Number: {self.department_number}\n"
                f"Date of Joining: {self.date_of_joining}")

def input_employee():
    employee_code = int(input("Enter employee code: "))
    name = input("Enter employee name: ")
    salary = float(input("Enter salary: "))
    department_number = int(input("Enter department number: "))
    day, month, year = map(int, input("Enter date of joining (day month year): ").split())
    date_of_joining = Date(day, month, year)
    return Employee(employee_code, name, salary, department_number, date_of_joining)

def main():
    employees = []
    
    # Input information for 10 employees
    for i in range(10):
        print(f"\nEnter information for employee {i + 1}:")
        employee = input_employee()
        employees.append(employee)
    
    # Display information for employees with salary >= 10000
    print("\nEmployees with salary >= 10000:")
    for emp in employees:
        if emp.salary >= 10000:
            print("\n" + str(emp))

if __name__ == "__main__":
    main()
